UI/window-basic-settings.cpp

```diff
@@ -1102,28 +1102,20 @@ void OBSBasicSettings::LoadGeneralSettings()
 
 	ui->multiviewLayout->addItem(QTStr(
 			"Basic.Settings.General.MultiviewLayout.Horizontal.Top"),
-			QT_UTF8("horizontaltop"));
+			static_cast<int>(MultiviewLayout::HORIZONTAL_TOP));
 	ui->multiviewLayout->addItem(QTStr(
 			"Basic.Settings.General.MultiviewLayout.Horizontal.Bottom"),
-			QT_UTF8("horizontalbottom"));
+			static_cast<int>(MultiviewLayout::HORIZONTAL_BOTTOM));
 	ui->multiviewLayout->addItem(QTStr(
 			"Basic.Settings.General.MultiviewLayout.Vertical.Left"),
-			QT_UTF8("verticalleft"));
+			static_cast<int>(MultiviewLayout::VERTICAL_LEFT));
 	ui->multiviewLayout->addItem(QTStr(
 			"Basic.Settings.General.MultiviewLayout.Vertical.Right"),
-			QT_UTF8("verticalright"));
+			static_cast<int>(MultiviewLayout::VERTICAL_RIGHT));
 
-	const char *multiviewLayoutText = config_get_string(GetGlobalConfig(),
-			"BasicWindow", "MultiviewLayout");
-
-	if (astrcmpi(multiviewLayoutText, "horizontalbottom") == 0)
-		ui->multiviewLayout->setCurrentIndex(1);
-	else if (astrcmpi(multiviewLayoutText, "verticalleft") == 0)
-		ui->multiviewLayout->setCurrentIndex(2);
-	else if (astrcmpi(multiviewLayoutText, "verticalright") == 0)
-		ui->multiviewLayout->setCurrentIndex(3);
-	else
-		ui->multiviewLayout->setCurrentIndex(0);
+	ui->multiviewLayout->setCurrentIndex(
+			config_get_int(GetGlobalConfig(), "BasicWindow",
+					"MultiviewLayout"));
```

Comment:
You're changing what a config file parameter expects here.  Previously, the [BasicWindow] MultiviewLayout variable was expected to be a string, and now you're sort of abruptly changing it to an int, basically breaking a user's existing config.

First it has to be noted that the reason why list index values aren't usually used as config values is because if you add/change something in that list box, it breaks configuration, hence why strings are typically used to represent their values.  Basically, it was done to prevent the chance of breaking configuration.

While an int is technically easier to use here, it's not necessarily safe due to that reason.

Changing it from expecting a string value to expecting an int value just breaks previous config values for this int, which is not typically a good practice to do.

1.) You can do this and allow it to break configuration
2.) You can use the string values instead.  There are probably easier ways to map string -> int values than this.  If possible, I would recommend to try to make it continue to use a string if it works with the current configuration.